superset-frontend/plugins/plugin-chart-cartodiagram/src/components/ChartLayer.tsx

```diff
@@ -192,9 +217,13 @@ export class ChartLayer extends Layer {
 
       const chartComponent = createChartComponent(
         this.chartVizType,
-        chart.feature,
+        // Seems like the registered chart components change some
+        // props in-place, which leads to unwanted side effects. Therefore,
+        // we only pass a copy.
+        JSON.parse(JSON.stringify(chart.feature)),
```

Comment:
minor: maybe lodash.clone(..) could be used

https://www.geeksforgeeks.org/lodash-_-clone-method/